### PR TITLE
feat: Up next button for recordings

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

For people binge watching recordings, a great feature is to be able to Netflix-style autocontinue

## Changes

* Adds a next up button when the video reaches the end
* Fixed some logic that determines whether we are at the end or not
* Refactors the types to be more concise
* Ended up also updating styles as they were borked for small screens

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## TODO:

- [x] Fix hotkeys not working...
- [x] Correct logic for restarting recording on play
- [x] Change hover logic to apply to the whole frame and not just the button
- [x] Add option to turn this off
- [x] Mount upcoming data logic for next recording in the next up button and then unmount it after a setTimeout

Follow up issue for later https://github.com/PostHog/posthog/issues/12768

### CODE REVIEW

File: frontend/src/scenes/session-recordings/playlist/sessionRecordingsListLogic.ts [hunk 1]
```diff
@@ -323,25 +332,20 @@ export const sessionRecordingsListLogic = kea<sessionRecordingsListLogicType>([
...
-                return
-            }
             const nulledSessionRecordingId = hashParams.sessionRecordingId ?? null
             if (nulledSessionRecordingId !== values.selectedRecordingId) {
                 actions.setSelectedRecordingId(nulledSessionRecordingId)
             }
 
             const filters = params.filters
-            if (filters) {
+            if (filters && props.updateSearchParams) {
```
Comment (benjackwhite) [inline frontend/src/scenes/session-recordings/playlist/sessionRecordingsListLogic.ts:341]: Noticed that we weren't persisting the hash param which we always want to for every playlist

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 2]
```diff
@@ -582,32 +611,6 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
                 cancelAnimationFrame(cache.timer)
             }
         },
-        handleKeyDown: ({ event }) => {
```
Comment (benjackwhite) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:585]: This wasn't working for some reason. Update it to use the more standard hooks variant

File: frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts [hunk 3]
```diff
@@ -353,12 +374,16 @@ export const sessionRecordingPlayerLogic = kea<sessionRecordingPlayerLogicType>(
...
+
             let nextPlayerPosition = values.currentPlayerPosition || values.currentSegment?.startPlayerPosition
 
-            if (values.currentSegment === values.sessionPlayerData.metadata.segments.slice(-1)[0]) {
-                // If we're at the end of the recording, go back to the beginning
+            if (values.endReached) {
                 nextPlayerPosition = values.sessionPlayerData.metadata.segments[0].startPlayerPosition
             }
+
+            console.log({ nextPlayerPosition, endReached: values.endReached })
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/player/sessionRecordingPlayerLogic.ts:384]: leftover console log

File: frontend/src/scenes/session-recordings/player/PlayerUpNext.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    const timeoutRef = useRef<any>()
+    const unmountNextRecordingDataLogicRef = useRef<() => void>()
+    const { endReached } = useValues(sessionRecordingPlayerLogic({ sessionRecordingId, playerKey }))
+    const { reportNextRecordingTriggered } = useActions(sessionRecordingPlayerLogic({ sessionRecordingId, playerKey }))
+    const [animate, setAnimate] = useState(false)
+    const { featureFlags } = useValues(featureFlagLogic)
+    const { autoplayEnabled } = useValues(playerSettingsLogic)
+
+    if (!autoplayEnabled || !featureFlags[FEATURE_FLAGS.RECORDING_AUTOPLAY]) {
+        nextSessionRecording = undefined
```
Comment (alexkim205) [inline frontend/src/scenes/session-recordings/player/PlayerUpNext.tsx:38]: Nit: don't mutate property

**Review by alexkim205 — APPROVED** [review]
Lgtm other than a few inline comments. One piece of feedback I have is that sometimes the minor fixes/refactors make it difficult to review the meat of the PR. Breaking out the refactoring changes into a separate PR would help a lot!

**Comment by benjackwhite** [comment]
> Lgtm other than a few inline comments. One piece of feedback I have is that sometimes the minor fixes/refactors make it difficult to review the meat of the PR. Breaking out the refactoring changes into a separate PR would help a lot!

Yeah thats really fair 😅 There's like 3 different chunks of work in here in one. Definitely will break it down next time 👍

**Pull request merged**